### PR TITLE
1941 do not use tc qualifier for pure cobol completion

### LINKED ISSUE #1941: Completion in pure COBOL should not use TC :: qualifier
### Description
In pure COBOL source code, the result of a completion should not use TypeCOBOL qualifier `::`, otherwise it won't compile.

Option `-cob` could be used to indicate that it's a pure COBOL context (building on #1927).

**sample code**
```cob
       IDENTIFICATION DIVISION.
       PROGRAM-ID. Pgm.
       DATA DIVISION.
       WORKING-STORAGE SECTION.
       01 Group1.
           05 var1        pic X.
           05 var2        pic X.
       PROCEDURE DIVISION.
      
              Display {cursor here}
           .
      
       END PROGRAM Pgm.
```
Asking for completion (ctrl+space) will display (`Group1`, `var1 (Group1::var1`, `Group1::var2`). When selecting var1 the resulting completion is `Group1::var1`.

Ideally, when asking for completion in a pure COBOL context every TypeCOBOL style qualifier `::` should be changed to Cobol85 `OF` operator, even for the suggestions listed when asking for completion, but this is out of scope for this issue.

### PR DESCRIPTION
**Pull request by fm-117**
Fixes #1941 

Minimal modification to the completion code to avoid `::` qualifier in variable completion suggestions. Does not affect procedures or variables from types that are always considered in a TypeCobol context.

### CODE REVIEW

**Review by rooksdo — APPROVED** [review]

**Review by delevoye — CHANGES_REQUESTED** [review]

File: TypeCobol.LanguageServer/Completion Factory/CompletionFactory.cs [hunk 1]
```diff
@@ -558,79 +559,99 @@ public static IEnumerable<CompletionItem> GetCompletionForTo(FileCompiler fileCo
...
+            foreach (var token in filteredTokens)
+            {
+                if (token ==  lastSignificantToken) break; //No need to go further
+                if (token == userFilterToken) continue;
+                switch (token.TokenType)
+                {
+                    case TokenType.QualifiedNameSeparator:
+                        continue;
+                    case TokenType.OF:
+                    case TokenType.IN:
```
[the remark this answers is not in the document]
Comment (fm-117) [reply TypeCobol.LanguageServer/Completion Factory/CompletionFactory.cs:623]: Yes, both can be used to separate parts of a qualified name.

File: TypeCobol.LanguageServer/Completion Factory/CompletionFactory.cs [hunk 2]
```diff
@@ -549,7 +550,7 @@ public static IEnumerable<CompletionItem> GetCompletionForVariable(FileCompiler
         #region TO Completion
         public static IEnumerable<CompletionItem> GetCompletionForTo(FileCompiler fileCompiler, CodeElement codeElement, Token userFilterToken, Token lastSignificantToken)
```
[the remark this answers is not in the document]
Comment (fm-117) [reply TypeCobol.LanguageServer/Completion Factory/CompletionFactory.cs:551]: Well, there are many improvements to make in those 2 files, but this PR is not meant to be a full refactoring of the completion mechanism. But I agree this part of LanguageServer needs a bit of code clean-up...

[the remark this answers is not in the document]
Comment (delevoye) [reply TypeCobol.LanguageServer/Completion Factory/CompletionFactory.cs:551]: Ok, looks like something to do in the future

File: TypeCobol.LanguageServer/Completion Factory/CompletionFactory.cs [hunk 3]
```diff
@@ -783,65 +800,61 @@ public static IReadOnlyList<Node> GetTypeChildren(SymbolTable symbolTable, DataD
...
-        /// Get the matchig node for a given Token and a gien completion mode. Returning a matching Node or null.
+        /// Get the matching node for the given CodeElement, returns null if not found.
         /// </summary>
-        /// <param name="fileCompiler"></param>
-        /// <param name="codeElement"></param>
-        /// <returns></returns>
+        /// <param name="fileCompiler">Current file being compiled with its compilation results</param>
+        /// <param name="codeElement">Target CodeElement</param>
+        /// <returns>Corresponding Node instance, null if not found.</returns>
         public static Node GetMatchingNode(FileCompiler fileCompiler, CodeElement codeElement)
```
[the remark this answers is not in the document]
Comment (fm-117) [reply TypeCobol.LanguageServer/Completion Factory/CompletionFactory.cs]: Kept the private ones in the main class, moved `GetMatchingNode`. See 48a294cca394f4a4ca362ecb198e5f3c23160d1d.

**Review by delevoye — APPROVED** [review]

**Review by mayanje — APPROVED** [review]

**Pull request merged**